presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddExchanges.java

```diff
@@ -988,7 +992,7 @@ public PlanWithProperties visitIndexJoin(IndexJoinNode node, PreferredProperties
             List<LocalProperty<VariableReferenceExpression>> desiredLocalProperties = preferredProperties.getLocalProperties().isEmpty() ? grouped(joinColumns) : ImmutableList.of();
 
             PlanWithProperties probeSource = node.getProbeSource().accept(this, PreferredProperties.partitionedWithLocal(ImmutableSet.copyOf(joinColumns), desiredLocalProperties)
-                    .mergeWithParent(preferredProperties));
+                    .mergeWithParent(preferredProperties, session));
```

Comment:
I'm not sure if it makes any practical sense to apply exact partitioning for IndexJoin, as the one doesn't use any memory. Maybe only just to fight the CPU skew?

Reply:
I'll remove the thee you called out.